### PR TITLE
 SapMachine (17) #1550: New malloc trace implementation

### PR DESCRIPTION
**Pull request by schmelter-sap**
Downport from PR 1640 (SapMachine21).

Needed fixes for command line handling and adjusted to the old jtreg test lib.

fixes #1550

### CODE REVIEW

**Comment by SapMachine** [comment]
Hello @schmelter-sap, this pull request fulfills all formal requirements.

**Comment by SapMachine** [comment]
Hello @schmelter-sap, this pull request fulfills all formal requirements.

**Comment by SapMachine** [comment]
Hello @schmelter-sap, this pull request fulfills all formal requirements.

**Comment by SapMachine** [comment]
Hello @schmelter-sap, this pull request fulfills all formal requirements.

**Comment by SapMachine** [comment]
Hello @schmelter-sap, this pull request fulfills all formal requirements.

**Review by RealCLanger — APPROVED** [review]
LGTM

File: test/hotspot/jtreg/runtime/malloctrace/MallocHooksTest.java [hunk 1]
```diff
@@ -0,0 +1,853 @@
...
+                    System.out.println("------- Repeating start of " + f.getName());
+                    System.out.println(output.substring(0, startLength));
+                    System.out.println("------- End of start of " + f.getName());
+                }
+            }
+        }
+    }
+
+    public static void main(String args[]) throws Exception {
+        if ((!Platform.isLinux() && !Platform.isOSX()) || !"64".equals(System.getProperty("sun.arch.data.model"))) {
```
Comment (reinrich) [inline test/hotspot/jtreg/runtime/malloctrace/MallocHooksTest.java]:
```suggestion
        if ((!Platform.isLinux() && !Platform.isOSX()) || Platform.is32bit()) {
```

Comment (schmelter-sap) [reply test/hotspot/jtreg/runtime/malloctrace/MallocHooksTest.java]: Good idea.

**Comment by SapMachine** [comment]
Hello @schmelter-sap, this pull request fulfills all formal requirements.

**Review by reinrich — APPROVED** [review]
The backport looks good to me.
Thanks, Richard.

**Pull request merged**